mkosi/__init__.py

```diff
@@ -2782,8 +2853,13 @@ def install_boot_loader(args: CommandLineArguments, root: str, loopdev: Optional
         return
 
     with complete_step("Installing boot loader"):
-        if args.esp_partno and args.distribution != Distribution.clear:
-            run_workspace_command(args, root, ["bootctl", "install"])
+        if args.esp_partno:
+            if args.distribution == Distribution.clear:
+                pass
+            elif args.distribution in (Distribution.centos, Distribution.centos_epel) and is_older_than_centos8(args.release):
+                install_boot_loader_centos_old_efi(args, root, loopdev)
```

Comment:
Not sure when I'll have spare time to implement this. For now I'd prefer to get the CentOS 7 fixes merged and maybe open an issue tracking that bootloader choice feature.
I might give it a try sometime in the future since I'll be maintaining our mkosi build pipeline, but no promises.

Reply:
Could you add the option and only implement it for centos 7? Early failure is fine on all the other distros. The idea behind that is that I want to keep the default behaviour for as much features as possible consistent across distros. Right now, uefi means systemd-boot for everything (except clear but that's a special one). I want to keep it that way ideally. Adding the bootloader option allows us to fail early for centos 7 if --bootloader grub is not specified so users have to add --bootloader grub explicitly for it to work.

Reply:
Also, I'm really interested in what you're using mkosi for. Can you shed any light on that?